tests/unit/deferred.py

```diff
@@ -96,17 +105,9 @@ def modify_port(self, port, channel, network_id):
 
             # setup a different connection to database so that this method does
             # not see uncommited changes by `apply_networking`
-
-            local_app = Flask(__name__.split('.')[0])
-            uri = config.cfg.get('database', 'uri')
-            local_app.config.update(SQLALCHEMY_TRACK_MODIFICATIONS=False)
-            local_app.config.update(SQLALCHEMY_DATABASE_URI=uri)
-            local_db = SQLAlchemy(local_app)
-
-            current_count = local_db.session \
-                .query(model.NetworkingAction).count()
-
-            local_db.session.commit()
+            with app.app_context():
```

Comment:
@zenhack - does this look right to you?

Skimming the docs again, I'm not sure whether app_context() or [scoped_session()](http://docs.sqlalchemy.org/en/latest/orm/contextual.html#unitofwork-contextual) is better.